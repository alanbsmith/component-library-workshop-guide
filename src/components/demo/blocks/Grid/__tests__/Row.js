import React from "react";
import Row from "../Row";
import renderer from "react-test-renderer";
import "jest-styled-components";

describe("Grid.Row Element", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Row />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("applies the correct style with the `position` modifier", () => {
    const tree = renderer.create(<Row modifiers={['positionRelative']} />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule("position", "relative");
  });
});
