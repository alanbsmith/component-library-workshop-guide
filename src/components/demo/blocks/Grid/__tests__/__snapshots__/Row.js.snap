// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid.Row Element applies the correct style with the \`position\` modifier 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
}

<div
  className="c0"
/>
`;

exports[`Grid.Row Element renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div
  className="c0"
/>
`;
